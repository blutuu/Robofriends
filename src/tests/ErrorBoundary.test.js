import React from "react";
import { shallow, mount } from "enzyme";
import ErrorBoundary from "../Components/ErrorBoundary";
import CardList from "../Components/CardList";

let wrapper = shallow(<ErrorBoundary />);
let wrapper2 = mount(
  <ErrorBoundary>
    <CardList />
  </ErrorBoundary>
);

describe("ErrorBoundary component testing", () => {
  it("renders the ErrorBoundary component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders the error message correctly", () => {
    // expect(wrapper.setState({ hasError: true })).toMatchSnapshot();
    expect(wrapper2).toMatchSnapshot();
  });
});
