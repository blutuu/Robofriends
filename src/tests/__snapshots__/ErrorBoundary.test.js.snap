// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component testing renders the ErrorBoundary component correctly 1`] = `""`;

exports[`ErrorBoundary component testing renders the error message correctly 1`] = `
<ErrorBoundary>
  <h1>
    Uh oh. An error occurred.
  </h1>
</ErrorBoundary>
`;
